src/geometry/euclidean/angle/unoriented/affine.lean

```diff
@@ -439,4 +439,51 @@ lemma angle_lt_pi_of_not_collinear {p₁ p₂ p₃ : P} (h : ¬collinear ℝ ({p
   ∠ p₁ p₂ p₃ < π :=
 (angle_le_pi _ _ _).lt_of_ne $ angle_ne_pi_of_not_collinear h
 
+/-- The cosine of the angle between three points is 1 if and only if the angle is 0. -/
+lemma cos_eq_one_iff_angle_eq_zero {p₁ p₂ p₃ : P} :
+  real.cos (∠ p₁ p₂ p₃) = 1 ↔ ∠ p₁ p₂ p₃ = 0 :=
+cos_eq_one_iff_angle_eq_zero _ _
+
+/-- The cosine of the angle between three points is 0 if and only if the angle is π / 2. -/
+lemma cos_eq_zero_iff_angle_eq_pi_div_two {p₁ p₂ p₃ : P} :
+  real.cos (∠ p₁ p₂ p₃) = 0 ↔ ∠ p₁ p₂ p₃ = π / 2 :=
+cos_eq_zero_iff_angle_eq_pi_div_two _ _
+
+/-- The cosine of the angle between three points is -1 if and only if the angle is π. -/
+lemma cos_eq_neg_one_iff_angle_eq_pi {p₁ p₂ p₃ : P} :
+  real.cos (∠ p₁ p₂ p₃) = -1 ↔ ∠ p₁ p₂ p₃ = π :=
+cos_eq_neg_one_iff_angle_eq_pi _ _
+
+/-- The sine of the angle between three points is 0 if and only if the angle is 0 or π. -/
+lemma sin_eq_zero_iff_angle_eq_zero_or_angle_eq_pi {p₁ p₂ p₃ : P} :
+  real.sin (∠ p₁ p₂ p₃) = 0 ↔ ∠ p₁ p₂ p₃ = 0 ∨ ∠ p₁ p₂ p₃ = π :=
+sin_eq_zero_iff_angle_eq_zero_or_angle_eq_pi _ _
+
+/-- The sine of the angle between three points is 1 if and only if the angle is π / 2. -/
+lemma sin_eq_one_iff_angle_eq_pi_div_two {p₁ p₂ p₃ : P} :
+  real.sin (∠ p₁ p₂ p₃) = 1 ↔ ∠ p₁ p₂ p₃ = π / 2 :=
+sin_eq_one_iff_angle_eq_pi_div_two _ _
```

Comment:
All of these lemmas are entirely redundant, I'd rather delete them.

Reply:
I don't think they're redundant. They're defeq to particular cases of the lemmas for angles between vectors that are used to prove them, but not syntactically equal; I don't think people wanting to do these rewrites for angles between points should have to rewrite by the definition of `euclidean_geometry.angle` and back again, as they would to use those lemmas for angles between vectors directly.

Reply:
This is analogous to the lemma's `angle_comm`, `angle_nonneg` and `anlge_le_pi` being stated for angles in vector form and in point angle form.

Reply:
A lot of the repetition might go away if `angle x y` and `∠ p q r` took values in `set.Icc 0 pi`, but that's out of scope for this PR.